[Bug]: Cannot find module 'next/dist/build/analysis/extract-const-value'

### Summary

Hey there,

We just ran a deploy after the latest plugin release and we are getting the following issue.

Error: Cannot find module 'next/dist/build/analysis/extract-const-value'
Require stack:

/opt/build/repo/web/.netlify/plugins/node_modules/@netlify/plugin-nextjs/lib/helpers/analysis.js
/opt/build/repo/web/.netlify/plugins/node_modules/@netlify/plugin-nextjs/lib/helpers/functions.js
/opt/build/repo/web/.netlify/plugins/node_modules/@netlify/plugin-nextjs/lib/index.js
While loading "@netlify/plugin-nextjs" from Netlify app at Function.Module._resolveFilename (node:internal/modules/cjs/loader:985:15) at Function.Module._load (node:internal/modules/cjs/loader:833:27) at Module.require (node:internal/modules/cjs/loader:1057:19) at require (node:internal/modules/cjs/helpers:103:18) at Object.<anonymous> (/opt/build/repo/web/.netlify/plugins/node_modules/@netlify/plugin-nextjs/lib/helpers/analysis.js:28:31) at Module._compile (node:internal/modules/cjs/loader:1155:14) at Object.Module._extensions..js (node:internal/modules/cjs/loader:1209:10) at Module.load (node:internal/modules/cjs/loader:1033:32) at Function.Module._load (node:internal/modules/cjs/loader:868:12) at Module.require (node:internal/modules/cjs/loader:1057:19)

Error properties: { code: 'MODULE_NOT_FOUND' }

### Steps to reproduce

1. Trigger Build

### A link to a reproduction repository

_No response_

### Next Runtime version

4.27.0

### More information about your build

- [ ] I am building using the CLI
- [ ] I am building using file-based configuration (`netlify.toml`)

### What OS are you using?

_No response_

### Your netlify.toml file

[build]
base = "/web/"
publish = ".next"
command = "yarn build"


### Your public/_redirects file

_No response_

### Your `next.config.js` file

// const { withSentryConfig } = require('@sentry/nextjs')

/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  images: {
    domains: ['cdn.sanity.io'],
    deviceSizes: [768, 1280, 1440, 1700],
    imageSizes: [45, 375, 710],
  },
  redirects: async () => [
    {
      source: '/',
      has: [
        {
          type: 'cookie',
          key: 'centreSlug',
          value: '(?<centreSlug>.*)',
        },
      ],
      destination: '/:centreSlug',
      permanent: false,
    },
    {
      source: '/',
      destination: '/intro',
      permanent: false,
    },
  ],
}
module.exports = nextConfig


### Builds logs (or link to your logs)

<details>
  <summary>Build logs</summary>

```
# Paste logs here
```

</details>


### Function logs

<details>
  <summary>Function logs</summary>

```
# Paste logs here
```

</details>


### .next JSON files

<details>
  <summary>generated .next JSON files</summary>

```
# Paste file contents here. Please check there isn't any private info in them
# You can either build locally, or download the deploy from Netlify by clicking the arrow next to the deploy time.
```

</details>

--- a/packages/runtime/src/helpers/analysis.ts
+++ b/packages/runtime/src/helpers/analysis.ts
@@ -1,7 +1,5 @@
 import fs, { existsSync } from 'fs'
 
-import { extractExportedConstValue, UnsupportedValueError } from 'next/dist/build/analysis/extract-const-value'
-import { parseModule } from 'next/dist/build/analysis/parse-module'
 import { relative } from 'pathe'
 
 // I have no idea what eslint is up to here but it gives an error
@@ -81,6 +79,9 @@ export const validateConfigValue = (config: ApiConfig, apiFilePath: string): con
   return false
 }
 
+let extractConstValue
+let parseModule
+let hasWarnedAboutNextVersion = false
 /**
  * Uses Next's swc static analysis to extract the config values from a file.
  */
@@ -88,6 +89,29 @@ export const extractConfigFromFile = async (apiFilePath: string): Promise<ApiCon
   if (!apiFilePath || !existsSync(apiFilePath)) {
     return {}
   }
+
+  try {
+    if (!extractConstValue) {
+      extractConstValue = require('next/dist/build/analysis/extract-const-value')
+    }
+    if (!parseModule) {
+      // eslint-disable-next-line prefer-destructuring, @typescript-eslint/no-var-requires
+      parseModule = require('next/dist/build/analysis/parse-module').parseModule
+    }
+  } catch (error) {
+    if (error.code === 'MODULE_NOT_FOUND') {
+      if (!hasWarnedAboutNextVersion) {
+        console.log("This version of Next.js doesn't support advanced API routes. Skipping...")
+        hasWarnedAboutNextVersion = true
+      }
+      // Old Next.js version
+      return {}
+    }
+    throw error
+  }
+
+  const { extractExportedConstValue, UnsupportedValueError } = extractConstValue
+
   const fileContent = await fs.promises.readFile(apiFilePath, 'utf8')
   // No need to parse if there's no "config"
   if (!fileContent.includes('config')) {
@@ -99,7 +123,7 @@ export const extractConfigFromFile = async (apiFilePath: string): Promise<ApiCon
   try {
     config = extractExportedConstValue(ast, 'config')
   } catch (error) {
-    if (error instanceof UnsupportedValueError) {
+    if (UnsupportedValueError && error instanceof UnsupportedValueError) {
       console.warn(`Unsupported config value in ${relative(process.cwd(), apiFilePath)}`)
     }
     return {}

